fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -1136,52 +1136,64 @@ std::vector<std::pair<std::string, bool>> MultiVersionApi::copyExternalLibraryPe
 	ASSERT_GE(threadCount, 1);
 	// Copy library for each thread configured per version
 	std::vector<std::pair<std::string, bool>> paths;
-	// It's tempting to use the so once without copying.  However, we don't know
-	// if the thing we're about to copy is the shared object executing this code
-	// or not, so this optimization is unsafe.
-	// paths.push_back({path, false});
-	for (int ii = 0; ii < threadCount; ++ii) {
-		std::string filename = basename(path);
-
-		char tempName[PATH_MAX + 12];
-		sprintf(tempName, "/tmp/%s-XXXXXX", filename.c_str());
-		int tempFd = mkstemp(tempName);
-		int fd;
-
-		if ((fd = open(path.c_str(), O_RDONLY)) == -1) {
-			TraceEvent("ExternalClientNotFound").detail("LibraryPath", path);
-			throw file_not_found();
-		}
 
-		constexpr size_t buf_sz = 4096;
-		char buf[buf_sz];
-		while (1) {
-			ssize_t readCount = read(fd, buf, buf_sz);
-			if (readCount == 0) {
-				// eof
-				break;
-			}
-			if (readCount == -1) {
-				TraceEvent(SevError, "ExternalClientCopyFailedReadError").GetLastError().detail("LibraryPath", path);
-				throw platform_error();
+	if (threadCount == 1) {
+		paths.push_back({ path, false });
+	} else {
+		// It's tempting to use the so once without copying. However, we don't know
+		// if the thing we're about to copy is the shared object executing this code
+		// or not, so this optimization is unsafe.
+		// paths.push_back({path, false});
+		for (int ii = 0; ii < threadCount; ++ii) {
+			std::string filename = basename(path);
+
+			char tempName[PATH_MAX + 12];
+			sprintf(tempName, "/tmp/%s-XXXXXX", filename.c_str());
+			int tempFd = mkstemp(tempName);
+			int fd;
+
+			if ((fd = open(path.c_str(), O_RDONLY)) == -1) {
+				TraceEvent("ExternalClientNotFound").detail("LibraryPath", path);
+				throw file_not_found();
 			}
-			ssize_t written = 0;
-			while (written != readCount) {
-				ssize_t writeCount = write(tempFd, buf + written, readCount - written);
-				if (writeCount == -1) {
-					TraceEvent(SevError, "ExternalClientCopyFailedWriteError")
+
+			TraceEvent("CopyingExternalClient")
```

Comment:
Git doesn't seem to have understood this change very well. Basically the entire previous block was indented and one new trace event was moved to this line from somewhere else.